lib/use_case/relationship/create.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "use_case/base"
+require "errors"
+
+module UseCase
+  module Relationship
+    class Create < UseCase::Base
+      option :current_user, type: Types::Coercible::String
+      option :target_user, type: Types::Coercible::String
+      option :type, type: Types::RelationshipTypes
+
+      def call
+        source_user = find_source_user
+        target_user = find_target_user
+
+        source_user.public_send(type, target_user)
+
+        {
+          status:   201,
+          resource: true,
+          extra:    {
+            target_user: target_user
+          }
+        }
+      end
+
+      private
+
+      def find_source_user
+        return current_user if current_user.is_a?(::User)
```

Comment:
`current_user` can never be a `::User` because of the type restriction

(side note: if the type is extended to also accept `::User` objects, you can pass in the `current_user` object directly to the UseCase call)